internal/api/list_detailed_projects_v1alpha1.go

```diff
@@ -0,0 +1,45 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"connectrpc.com/connect"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	svcv1alpha1 "github.com/akuity/kargo/pkg/api/service/v1alpha1"
+)
+
+func (s *server) ListDetailedProjects(
+	ctx context.Context,
+	_ *connect.Request[svcv1alpha1.ListDetailedProjectsRequest],
+) (*connect.Response[svcv1alpha1.ListDetailedProjectsResponse], error) {
+	var list kargoapi.ProjectList
+	if err := s.client.List(ctx, &list); err != nil {
+		return nil, fmt.Errorf("error listing Projects: %w", err)
+	}
+
+	sort.Slice(list.Items, func(i, j int) bool {
```

Comment:
We should start making use of the [standard `slices` package](https://pkg.go.dev/slices) which is faster in many cases. I think there are examples elsewhere in the code base for lexicographic sorting.